design/fips-compliance.md

```diff
@@ -0,0 +1,293 @@
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | <ul><li>Customers who use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`</li></ul>                                                                                                                                                                                                      |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test and document. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+Note: If the performance of the `goboring/golang` version will be much worse than that
+      of the `golang` version then we will release both versions.
+      More info on the required performance, and the performance tests that will be done in this effort can be found in the
+      [Performance section](#performance)
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. As with the regular image, we will replace the
+current one with a FIPS-compliant image.
+
+### Performance
+
+We should verify that the performance of the FIPS-compliant image doesn't 
+have a degradation that is "visible to the naked eye". We will deploy a pod with
+an init container of the Non-Fips image, and then deploy a pod with an 
+init container of the Fips image. If the difference is a matter of seconds then
+the performance is good. If the difference is a matter of minutes then the performance of the 
+FIPS-compliant image is not good.
```

Comment:
I think that the first phase is to run with FIPS understand if we have a performance issue, only if we feel that there is some degredation we should compare to non-FIPS.
this is IMO - degradation that is "visible to the naked eye"

Reply:
how can we understand if we have a performance issue if we don't compare it with the non-FIPS version?

Reply:
we should see if it doesn't take too much time to initialize, if it feels like ages we should compare it, but if it takes in no time we shouldn't.
I believe that since you use it for a while now, it should be obvious if it takes more or less the same or much more than it should.